src/gobstuf/api.py

```diff
@@ -105,28 +124,31 @@ def _stuf():
     audit_logger = AuditLogger.get_instance()
 
     request = flask.request
-
-    method = request.method
-    assert method in ['GET', 'POST'], f"Unknown method {method}, GET or POST required"
-
     url = _routed_url(request.url)
-    if method == 'GET':
-        response = _get_stuf(url)
-    elif method == 'POST':
-        data = _update_request(request.data.decode())
-        response = _post_stuf(url, data, request.headers)
 
-    text = response.text
-    text = _update_response(text)
+    request_uuid = str(uuid.uuid4())
 
-    extra_data = {
+    request_log_data = {
         'soapaction': request.headers.get('Soapaction'),
-        'remote_response_code': response.status_code,
         'original_url': request.url,
-        'method': method,
+        'method': request.method,
     }
-
-    audit_logger.log_request(request.remote_addr, url, extra_data)
+    audit_logger.log_request(request.remote_addr, url, request_log_data, request_uuid)
+
+    response_log_data = {**request_log_data}
+
+    try:
+        response = _handle_stuf_request(request, url)
+    except HTTPException as e:
+        # If Exception occurs, log exception and re-raise
```

Comment:
Just a question: is there any chance that the address of the remote soap service (MKS) will be visible to the caller?
If for instance the call to MKS fails, is there any chance that the exception message that is received by the caller reveals the address of MKS? This should never happen.

Reply:
No. We're raising the exceptions ourselves. HTTPException is a Flask (Werkzeug) exception class. These exceptions are never raised at that level, except when we do so.

Reply:
OK, so it's safe. Thanks